examples/dnsdoctor/dnsdoctor.lua

```diff
@@ -0,0 +1,77 @@
+--
+-- SPDX-FileCopyrightText: (c) 2024 Mohammad Shehar Yaar Tausif <sheharyaar48@gmail.com>
+-- SPDX-License-Identifier: MIT OR GPL-2.0-only
+--
+
+-- DNS Doctoring : Rewrite DNS type A record to a private address for local clients
+
+local xt = require("xtable")
+local linux = require("linux")
+local string = require("string")
+local action = xt.action
+local family = xt.family
+
+local udp = 0x11
+local dns = 0x35
+
+local function nop() end
+
+local function get_domain(skb, off)
+	local name = ""
+	local i = 0
+
+	repeat
+		local d = skb:getbyte(off + i)
+		i = i + 1
+	until d == 0
+
+	return name .. skb:getstring(off, i-1), i
+end
+
+local function dnsdoctor_tg(skb, par, userdata)
```

Comment:
I think userdata here is misleading.. because it corresponds to a Lua type name.. I would `arg` or something else..